### DISCUSSION TITLE
I'm a noob and I want to know if it is possible to authenticate with SAML 2.0 without input from user

### DISCUSSION THREAD

**Discussion by fwehuerguweioergjioerier** [keycloak/keycloak#13754] [Q&A]
Hello, I have a application that supports SAML 2.0 for authentication. (Rocket.Chat & TopDesk)

I want my users to authenticate in Windows in the morning and all other programs use the credentials from the AD account without the user having to enter them again.
Is it possible with Keycloak to have the tool grab the credentials from the logged in user and provide them to the SAML interface in for example Rocket.Chat? 

Thanks for help!

**Comment by Aleson95** [comment]
I'm not sure about "Keycloak to have the tool grab the credentials from the logged in user" is possible or not
but keycloak provides User Federation function for us to import user data from our define LDAP system such as Microsoft AD.
From there your keycloak should have all the user you want defined by your LDAP query, and then you can configure Keycloak as (IDP) and import the SAML clients (SP), after that you can use keycloak to login to your SAML clients.
